controllers/operator_controller.go

```diff
@@ -176,20 +210,93 @@ func (r *OperatorReconciler) reconcile(ctx context.Context, op *operatorsv1alpha
 	}
 
+	// Let's set the proper Installed condition and installedBundleSource field based on the
+	// existing BundleDeployment object status.
+	installedCond, installedBundleSource := mapBDStatusToInstalledCondition(existingTypedBundleDeployment, op)
+	apimeta.SetStatusCondition(&op.Status.Conditions, installedCond)
+	op.Status.InstalledBundleSource = installedBundleSource
+
 	// set the status of the operator based on the respective bundle deployment status conditions.
 	return ctrl.Result{}, nil
 }
 
+func mapBDStatusToInstalledCondition(existingTypedBundleDeployment *rukpakv1alpha1.BundleDeployment, op *operatorsv1alpha1.Operator) (metav1.Condition, string) {
+	bundleDeploymentReady := apimeta.FindStatusCondition(existingTypedBundleDeployment.Status.Conditions, rukpakv1alpha1.TypeInstalled)
+	if bundleDeploymentReady == nil {
+		return metav1.Condition{
+			Type:               operatorsv1alpha1.TypeInstalled,
+			Status:             metav1.ConditionUnknown,
+			Reason:             operatorsv1alpha1.ReasonInstallationStatusUnknown,
+			Message:            "bundledeployment status is unknown",
+			ObservedGeneration: op.GetGeneration(),
+		}, ""
+	}
+
+	if bundleDeploymentReady.Status != metav1.ConditionTrue {
+		return metav1.Condition{
+			Type:               operatorsv1alpha1.TypeInstalled,
+			Status:             metav1.ConditionFalse,
+			Reason:             operatorsv1alpha1.ReasonInstallationFailed,
+			Message:            fmt.Sprintf("bundledeployment not ready: %s", bundleDeploymentReady.Message),
+			ObservedGeneration: op.GetGeneration(),
+		}, ""
+	}
+
+	bundleDeploymentSource := existingTypedBundleDeployment.Spec.Template.Spec.Source
+	switch bundleDeploymentSource.Type {
+	case rukpakv1alpha1.SourceTypeImage:
+		return metav1.Condition{
+			Type:               operatorsv1alpha1.TypeInstalled,
+			Status:             metav1.ConditionTrue,
+			Reason:             operatorsv1alpha1.ReasonSuccess,
+			Message:            fmt.Sprintf("installed from %q", bundleDeploymentSource.Image.Ref),
+			ObservedGeneration: op.GetGeneration(),
+		}, bundleDeploymentSource.Image.Ref
+	case rukpakv1alpha1.SourceTypeGit:
+		return metav1.Condition{
+			Type:               operatorsv1alpha1.TypeInstalled,
+			Status:             metav1.ConditionTrue,
+			Reason:             operatorsv1alpha1.ReasonSuccess,
+			Message:            fmt.Sprintf("installed from %q", bundleDeploymentSource.Git.Repository+"@"+bundleDeploymentSource.Git.Ref.Commit),
+			ObservedGeneration: op.GetGeneration(),
+		}, bundleDeploymentSource.Git.Repository + "@" + bundleDeploymentSource.Git.Ref.Commit
+	default:
+		return metav1.Condition{
+			Type:               operatorsv1alpha1.TypeInstalled,
+			Status:             metav1.ConditionUnknown,
+			Reason:             operatorsv1alpha1.ReasonInstallationStatusUnknown,
+			Message:            fmt.Sprintf("unknown bundledeployment source type %q", bundleDeploymentSource.Type),
+			ObservedGeneration: op.GetGeneration(),
+		}, ""
```

Comment:
Nit: 

Coming into this a bit unfamiliar with the actual code base, but from what I can tell this is the 7th use of creating this condition with only the `Condition.Message` field being different. It might be nice to create a function like:
```go
func installStatusUnknownCondition(msg string, generation int64) metav1.Condition {
  return metav1.Condition{
 	Type:               operatorsv1alpha1.TypeInstalled,
 	Status:             metav1.ConditionUnknown,
 	Reason:             operatorsv1alpha1.ReasonInstallationStatusUnknown,
 	Message:            msg,
 	ObservedGeneration: generation,
 }
}
```

I don't know if this is a followed pattern in the code base, but IMO this causes there to be a bit less bloat to read through when looking at the code and it will make it so you only have to change a few lines if in the future we wanted to tweak the values for this condition at all

Reply:
I don't have a strong opinion on this. Still, after trying the code change, I'm unsure about only creating a helper for the InstallStatus Unknown condition, as the code looks "weird/harder to read" having some conditions declared differently.

How is this usually solved in other codebases?

Reply:
> having some conditions declared differently.
I think this is the biggest thing about *only* doing this for the one condition. IIRC both catalogd and rukpak use helper functions for *all* the status conditions, so I'd be fine with this being a follow up item to clean up the code for all the status conditions

Reply:
@everettraven makes sense; I will start a PR to cleanup this code by creating the proper helpers for all the conditions.